sql/tests: TestRandomSyntaxFunctions failed

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11186098?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11186098?buildTab=artifacts#/) on master @ [d87a04bf528cf3d8f242ceb539451cd14aa7a3cf](https://github.com/cockroachdb/cockroach/commits/d87a04bf528cf3d8f242ceb539451cd14aa7a3cf):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 5248334 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:844: 40s of 5m0s: 127091 executions, 78470 successful
    rsg_test.go:844: 45s of 5m0s: 143814 executions, 88784 successful
    rsg_test.go:844: 50s of 5m0s: 160175 executions, 98944 successful
    rsg_test.go:844: 55s of 5m0s: 175287 executions, 108265 successful
    rsg_test.go:844: 1m0s of 5m0s: 190104 executions, 117518 successful
    rsg_test.go:844: 1m5s of 5m0s: 201785 executions, 124717 successful
    rsg_test.go:844: 1m10s of 5m0s: 217913 executions, 134684 successful
    rsg_test.go:844: 1m15s of 5m0s: 234667 executions, 145104 successful
    rsg_test.go:844: 1m20s of 5m0s: 251241 executions, 155315 successful
    rsg_test.go:844: 1m25s of 5m0s: 265669 executions, 164301 successful
    rsg_test.go:844: 1m30s of 5m0s: 282359 executions, 174531 successful
    rsg_test.go:844: 1m35s of 5m0s: 297886 executions, 184177 successful
    rsg_test.go:844: 1m40s of 5m0s: 314521 executions, 194583 successful
    rsg_test.go:844: 1m45s of 5m0s: 330626 executions, 204521 successful
    rsg_test.go:844: 1m50s of 5m0s: 347220 executions, 214735 successful
    rsg_test.go:844: 1m55s of 5m0s: 360902 executions, 223159 successful
    rsg_test.go:844: 2m0s of 5m0s: 378009 executions, 233716 successful
    rsg_test.go:844: 2m5s of 5m0s: 394719 executions, 244161 successful
    rsg_test.go:844: 2m10s of 5m0s: 411398 executions, 254417 successful
    rsg_test.go:844: 2m15s of 5m0s: 427820 executions, 264590 successful
    rsg_test.go:844: 2m20s of 5m0s: 443526 executions, 274305 successful
    rsg_test.go:844: 2m25s of 5m0s: 460388 executions, 284631 successful
    rsg_test.go:844: 2m30s of 5m0s: 475000 executions, 293728 successful
    rsg_test.go:844: 2m35s of 5m0s: 491374 executions, 303811 successful
    rsg_test.go:844: 2m40s of 5m0s: 507595 executions, 313871 successful
    rsg_test.go:844: 2m45s of 5m0s: 523800 executions, 323922 successful
    rsg_test.go:844: 2m50s of 5m0s: 540216 executions, 334072 successful
    rsg_test.go:844: 2m55s of 5m0s: 554863 executions, 343224 successful
    rsg_test.go:844: 3m0s of 5m0s: 570725 executions, 353014 successful
    rsg_test.go:844: 3m5s of 5m0s: 587104 executions, 363101 successful
    rsg_test.go:844: 3m10s of 5m0s: 603748 executions, 373450 successful
    rsg_test.go:844: 3m15s of 5m0s: 620405 executions, 383737 successful
    rsg_test.go:844: 3m20s of 5m0s: 636922 executions, 394020 successful
    rsg_test.go:844: 3m25s of 5m0s: 654318 executions, 404828 successful
    rsg_test.go:844: 3m30s of 5m0s: 668750 executions, 413590 successful
    rsg_test.go:844: 3m35s of 5m0s: 684628 executions, 423435 successful
    rsg_test.go:844: 3m40s of 5m0s: 700865 executions, 433512 successful
    rsg_test.go:844: 3m45s of 5m0s: 717402 executions, 443667 successful
    rsg_test.go:844: 3m50s of 5m0s: 734122 executions, 454143 successful
    rsg_test.go:844: 3m55s of 5m0s: 748763 executions, 463155 successful
    rsg_test.go:844: 4m0s of 5m0s: 765838 executions, 473731 successful
    rsg_test.go:844: 4m5s of 5m0s: 782688 executions, 484070 successful
    rsg_test.go:844: 4m10s of 5m0s: 798308 executions, 493769 successful
    rsg_test.go:844: 4m15s of 5m0s: 815614 executions, 504508 successful
    rsg_test.go:844: 4m20s of 5m0s: 831493 executions, 514275 successful
    rsg_test.go:844: 4m25s of 5m0s: 846131 executions, 523397 successful
    rsg_test.go:844: 4m30s of 5m0s: 862898 executions, 533793 successful
    rsg_test.go:844: 4m35s of 5m0s: 878427 executions, 543377 successful
    rsg_test.go:844: 4m40s of 5m0s: 894687 executions, 553396 successful
    rsg_test.go:844: 4m45s of 5m0s: 911702 executions, 563867 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #107928 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-23.1]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30332

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11200334?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11200334?buildTab=artifacts#/) on master @ [695d4ec85157a4da0c97237e6f16bf919eebf6c5](https://github.com/cockroachdb/cockroach/commits/695d4ec85157a4da0c97237e6f16bf919eebf6c5):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 3908174 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:844: 40s of 5m0s: 124006 executions, 76806 successful
    rsg_test.go:844: 45s of 5m0s: 140632 executions, 87081 successful
    rsg_test.go:844: 50s of 5m0s: 156585 executions, 97040 successful
    rsg_test.go:844: 55s of 5m0s: 172464 executions, 106872 successful
    rsg_test.go:844: 1m0s of 5m0s: 184676 executions, 114382 successful
    rsg_test.go:844: 1m5s of 5m0s: 196199 executions, 121475 successful
    rsg_test.go:844: 1m10s of 5m0s: 210970 executions, 130641 successful
    rsg_test.go:844: 1m15s of 5m0s: 226947 executions, 140605 successful
    rsg_test.go:844: 1m20s of 5m0s: 242463 executions, 150176 successful
    rsg_test.go:844: 1m25s of 5m0s: 258439 executions, 160023 successful
    rsg_test.go:844: 1m30s of 5m0s: 272758 executions, 168880 successful
    rsg_test.go:844: 1m35s of 5m0s: 289312 executions, 179168 successful
    rsg_test.go:844: 1m40s of 5m0s: 305193 executions, 188983 successful
    rsg_test.go:844: 1m45s of 5m0s: 319885 executions, 198029 successful
    rsg_test.go:844: 1m50s of 5m0s: 336350 executions, 208276 successful
    rsg_test.go:844: 1m55s of 5m0s: 350793 executions, 217171 successful
    rsg_test.go:844: 2m0s of 5m0s: 366779 executions, 226988 successful
    rsg_test.go:844: 2m5s of 5m0s: 382771 executions, 236934 successful
    rsg_test.go:844: 2m10s of 5m0s: 398300 executions, 246512 successful
    rsg_test.go:844: 2m15s of 5m0s: 414418 executions, 256462 successful
    rsg_test.go:844: 2m20s of 5m0s: 429914 executions, 266044 successful
    rsg_test.go:844: 2m25s of 5m0s: 444257 executions, 274857 successful
    rsg_test.go:844: 2m30s of 5m0s: 460349 executions, 284826 successful
    rsg_test.go:844: 2m35s of 5m0s: 476337 executions, 294720 successful
    rsg_test.go:844: 2m40s of 5m0s: 493000 executions, 305032 successful
    rsg_test.go:844: 2m45s of 5m0s: 508367 executions, 314605 successful
    rsg_test.go:844: 2m50s of 5m0s: 524752 executions, 324759 successful
    rsg_test.go:844: 2m55s of 5m0s: 539141 executions, 333792 successful
    rsg_test.go:844: 3m0s of 5m0s: 555226 executions, 343621 successful
    rsg_test.go:844: 3m5s of 5m0s: 570679 executions, 353186 successful
    rsg_test.go:844: 3m10s of 5m0s: 586571 executions, 362935 successful
    rsg_test.go:844: 3m15s of 5m0s: 603003 executions, 373137 successful
    rsg_test.go:844: 3m20s of 5m0s: 616614 executions, 381487 successful
    rsg_test.go:844: 3m25s of 5m0s: 633117 executions, 391769 successful
    rsg_test.go:844: 3m30s of 5m0s: 648725 executions, 401386 successful
    rsg_test.go:844: 3m35s of 5m0s: 665254 executions, 411523 successful
    rsg_test.go:844: 3m40s of 5m0s: 681328 executions, 421382 successful
    rsg_test.go:844: 3m45s of 5m0s: 697100 executions, 431184 successful
    rsg_test.go:844: 3m50s of 5m0s: 712957 executions, 440998 successful
    rsg_test.go:844: 3m55s of 5m0s: 726726 executions, 449470 successful
    rsg_test.go:844: 4m0s of 5m0s: 743390 executions, 459732 successful
    rsg_test.go:844: 4m5s of 5m0s: 758531 executions, 469105 successful
    rsg_test.go:844: 4m10s of 5m0s: 774146 executions, 478758 successful
    rsg_test.go:844: 4m15s of 5m0s: 790604 executions, 488988 successful
    rsg_test.go:844: 4m20s of 5m0s: 805137 executions, 497916 successful
    rsg_test.go:844: 4m25s of 5m0s: 821003 executions, 507687 successful
    rsg_test.go:844: 4m30s of 5m0s: 836112 executions, 517124 successful
    rsg_test.go:844: 4m35s of 5m0s: 852345 executions, 527062 successful
    rsg_test.go:844: 4m40s of 5m0s: 867856 executions, 536616 successful
    rsg_test.go:844: 4m45s of 5m0s: 884037 executions, 546617 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #107928 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-23.1]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11207252?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11207252?buildTab=artifacts#/) on master @ [f1623e0a2cffc81580ee3c3b71035efd854d89c9](https://github.com/cockroachdb/cockroach/commits/f1623e0a2cffc81580ee3c3b71035efd854d89c9):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 3859506 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:844: 40s of 5m0s: 123746 executions, 76451 successful
    rsg_test.go:844: 45s of 5m0s: 140136 executions, 86574 successful
    rsg_test.go:844: 50s of 5m0s: 156106 executions, 96429 successful
    rsg_test.go:844: 55s of 5m0s: 170461 executions, 105386 successful
    rsg_test.go:844: 1m0s of 5m0s: 184691 executions, 114109 successful
    rsg_test.go:844: 1m5s of 5m0s: 196359 executions, 121244 successful
    rsg_test.go:844: 1m10s of 5m0s: 211881 executions, 130838 successful
    rsg_test.go:844: 1m15s of 5m0s: 228102 executions, 140912 successful
    rsg_test.go:844: 1m20s of 5m0s: 242727 executions, 149992 successful
    rsg_test.go:844: 1m25s of 5m0s: 258738 executions, 159838 successful
    rsg_test.go:844: 1m30s of 5m0s: 274916 executions, 169834 successful
    rsg_test.go:844: 1m35s of 5m0s: 290927 executions, 179724 successful
    rsg_test.go:844: 1m40s of 5m0s: 307162 executions, 189837 successful
    rsg_test.go:844: 1m45s of 5m0s: 323814 executions, 200117 successful
    rsg_test.go:844: 1m50s of 5m0s: 338332 executions, 209192 successful
    rsg_test.go:844: 1m55s of 5m0s: 354520 executions, 219155 successful
    rsg_test.go:844: 2m0s of 5m0s: 370440 executions, 229055 successful
    rsg_test.go:844: 2m5s of 5m0s: 385927 executions, 238654 successful
    rsg_test.go:844: 2m10s of 5m0s: 402347 executions, 248859 successful
    rsg_test.go:844: 2m15s of 5m0s: 416507 executions, 257655 successful
    rsg_test.go:844: 2m20s of 5m0s: 431791 executions, 267160 successful
    rsg_test.go:844: 2m25s of 5m0s: 448011 executions, 277084 successful
    rsg_test.go:844: 2m30s of 5m0s: 464424 executions, 287284 successful
    rsg_test.go:844: 2m35s of 5m0s: 480877 executions, 297501 successful
    rsg_test.go:844: 2m40s of 5m0s: 497346 executions, 307638 successful
    rsg_test.go:844: 2m45s of 5m0s: 512077 executions, 316817 successful
    rsg_test.go:844: 2m50s of 5m0s: 527885 executions, 326509 successful
    rsg_test.go:844: 2m55s of 5m0s: 544467 executions, 336744 successful
    rsg_test.go:844: 3m0s of 5m0s: 560753 executions, 346877 successful
    rsg_test.go:844: 3m5s of 5m0s: 576534 executions, 356581 successful
    rsg_test.go:844: 3m10s of 5m0s: 593091 executions, 366721 successful
    rsg_test.go:844: 3m15s of 5m0s: 607075 executions, 375295 successful
    rsg_test.go:844: 3m20s of 5m0s: 623189 executions, 385307 successful
    rsg_test.go:844: 3m25s of 5m0s: 639453 executions, 395317 successful
    rsg_test.go:844: 3m30s of 5m0s: 655466 executions, 405244 successful
    rsg_test.go:844: 3m35s of 5m0s: 671778 executions, 415389 successful
    rsg_test.go:844: 3m40s of 5m0s: 687641 executions, 425181 successful
    rsg_test.go:844: 3m45s of 5m0s: 704337 executions, 435455 successful
    rsg_test.go:844: 3m50s of 5m0s: 718061 executions, 443951 successful
    rsg_test.go:844: 3m55s of 5m0s: 733150 executions, 453268 successful
    rsg_test.go:844: 4m0s of 5m0s: 748970 executions, 463089 successful
    rsg_test.go:844: 4m5s of 5m0s: 765036 executions, 473049 successful
    rsg_test.go:844: 4m10s of 5m0s: 780868 executions, 482798 successful
    rsg_test.go:844: 4m15s of 5m0s: 797207 executions, 492967 successful
    rsg_test.go:844: 4m20s of 5m0s: 811547 executions, 501795 successful
    rsg_test.go:844: 4m25s of 5m0s: 827310 executions, 511585 successful
    rsg_test.go:844: 4m30s of 5m0s: 843250 executions, 521404 successful
    rsg_test.go:844: 4m35s of 5m0s: 859434 executions, 531524 successful
    rsg_test.go:844: 4m40s of 5m0s: 875847 executions, 541533 successful
    rsg_test.go:844: 4m45s of 5m0s: 891630 executions, 551282 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #107928 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-23.1]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11213409?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11213409?buildTab=artifacts#/) on master @ [f1623e0a2cffc81580ee3c3b71035efd854d89c9](https://github.com/cockroachdb/cockroach/commits/f1623e0a2cffc81580ee3c3b71035efd854d89c9):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 5725711 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:844: 45s of 5m0s: 143826 executions, 89083 successful
    rsg_test.go:844: 50s of 5m0s: 160805 executions, 99668 successful
    rsg_test.go:844: 55s of 5m0s: 177292 executions, 109814 successful
    rsg_test.go:844: 1m0s of 5m0s: 191056 executions, 118343 successful
    rsg_test.go:844: 1m5s of 5m0s: 204650 executions, 126725 successful
    rsg_test.go:844: 1m10s of 5m0s: 221075 executions, 136940 successful
    rsg_test.go:844: 1m15s of 5m0s: 237827 executions, 147218 successful
    rsg_test.go:844: 1m20s of 5m0s: 254329 executions, 157505 successful
    rsg_test.go:844: 1m25s of 5m0s: 268997 executions, 166565 successful
    rsg_test.go:844: 1m30s of 5m0s: 285693 executions, 176986 successful
    rsg_test.go:844: 1m35s of 5m0s: 302643 executions, 187482 successful
    rsg_test.go:844: 1m40s of 5m0s: 318700 executions, 197431 successful
    rsg_test.go:844: 1m45s of 5m0s: 335217 executions, 207599 successful
    rsg_test.go:844: 1m50s of 5m0s: 349923 executions, 216736 successful
    rsg_test.go:844: 1m55s of 5m0s: 366706 executions, 227082 successful
    rsg_test.go:844: 2m0s of 5m0s: 383730 executions, 237582 successful
    rsg_test.go:844: 2m5s of 5m0s: 399772 executions, 247596 successful
    rsg_test.go:844: 2m10s of 5m0s: 416217 executions, 257821 successful
    rsg_test.go:844: 2m15s of 5m0s: 431820 executions, 267517 successful
    rsg_test.go:844: 2m20s of 5m0s: 445796 executions, 276131 successful
    rsg_test.go:844: 2m25s of 5m0s: 462219 executions, 286236 successful
    rsg_test.go:844: 2m30s of 5m0s: 477701 executions, 295833 successful
    rsg_test.go:844: 2m35s of 5m0s: 494212 executions, 306098 successful
    rsg_test.go:844: 2m40s of 5m0s: 510986 executions, 316406 successful
    rsg_test.go:844: 2m45s of 5m0s: 527448 executions, 326695 successful
    rsg_test.go:844: 2m50s of 5m0s: 541774 executions, 335455 successful
    rsg_test.go:844: 2m55s of 5m0s: 558765 executions, 346073 successful
    rsg_test.go:844: 3m0s of 5m0s: 575069 executions, 356093 successful
    rsg_test.go:844: 3m5s of 5m0s: 591694 executions, 366442 successful
    rsg_test.go:844: 3m10s of 5m0s: 608465 executions, 376768 successful
    rsg_test.go:844: 3m15s of 5m0s: 624823 executions, 386943 successful
    rsg_test.go:844: 3m20s of 5m0s: 642160 executions, 397686 successful
    rsg_test.go:844: 3m25s of 5m0s: 655802 executions, 406174 successful
    rsg_test.go:844: 3m30s of 5m0s: 672591 executions, 416587 successful
    rsg_test.go:844: 3m35s of 5m0s: 688546 executions, 426553 successful
    rsg_test.go:844: 3m40s of 5m0s: 705133 executions, 436944 successful
    rsg_test.go:844: 3m45s of 5m0s: 721961 executions, 447292 successful
    rsg_test.go:844: 3m50s of 5m0s: 738411 executions, 457583 successful
    rsg_test.go:844: 3m55s of 5m0s: 753223 executions, 466750 successful
    rsg_test.go:844: 4m0s of 5m0s: 769928 executions, 477111 successful
    rsg_test.go:844: 4m5s of 5m0s: 786402 executions, 487313 successful
    rsg_test.go:844: 4m10s of 5m0s: 802939 executions, 497629 successful
    rsg_test.go:844: 4m15s of 5m0s: 819992 executions, 508154 successful
    rsg_test.go:844: 4m20s of 5m0s: 836650 executions, 518474 successful
    rsg_test.go:844: 4m25s of 5m0s: 850795 executions, 527216 successful
    rsg_test.go:844: 4m30s of 5m0s: 867551 executions, 537648 successful
    rsg_test.go:844: 4m35s of 5m0s: 885065 executions, 548612 successful
    rsg_test.go:844: 4m40s of 5m0s: 901446 executions, 558678 successful
    rsg_test.go:844: 4m45s of 5m0s: 917650 executions, 568699 successful
    rsg_test.go:844: 4m50s of 5m0s: 934970 executions, 579554 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #107928 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-23.1]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11229029?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11229029?buildTab=artifacts#/) on master @ [1382b26a97bf6f70a07d363dc319283c173359eb](https://github.com/cockroachdb/cockroach/commits/1382b26a97bf6f70a07d363dc319283c173359eb):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 5500644 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:844: 40s of 5m0s: 119808 executions, 74057 successful
    rsg_test.go:844: 45s of 5m0s: 135451 executions, 83675 successful
    rsg_test.go:844: 50s of 5m0s: 151397 executions, 93458 successful
    rsg_test.go:844: 55s of 5m0s: 165954 executions, 102501 successful
    rsg_test.go:844: 1m0s of 5m0s: 180748 executions, 111611 successful
    rsg_test.go:844: 1m5s of 5m0s: 190973 executions, 117962 successful
    rsg_test.go:844: 1m10s of 5m0s: 204814 executions, 126575 successful
    rsg_test.go:844: 1m15s of 5m0s: 220129 executions, 136165 successful
    rsg_test.go:844: 1m20s of 5m0s: 234942 executions, 145336 successful
    rsg_test.go:844: 1m25s of 5m0s: 250467 executions, 154933 successful
    rsg_test.go:844: 1m30s of 5m0s: 266137 executions, 164608 successful
    rsg_test.go:844: 1m35s of 5m0s: 279188 executions, 172653 successful
    rsg_test.go:844: 1m40s of 5m0s: 293980 executions, 181763 successful
    rsg_test.go:844: 1m45s of 5m0s: 310037 executions, 191732 successful
    rsg_test.go:844: 1m50s of 5m0s: 325665 executions, 201439 successful
    rsg_test.go:844: 1m55s of 5m0s: 341138 executions, 211024 successful
    rsg_test.go:844: 2m0s of 5m0s: 355496 executions, 219924 successful
    rsg_test.go:844: 2m5s of 5m0s: 370307 executions, 229003 successful
    rsg_test.go:844: 2m10s of 5m0s: 386018 executions, 238747 successful
    rsg_test.go:844: 2m15s of 5m0s: 401646 executions, 248408 successful
    rsg_test.go:844: 2m20s of 5m0s: 417336 executions, 258116 successful
    rsg_test.go:844: 2m25s of 5m0s: 432625 executions, 267478 successful
    rsg_test.go:844: 2m30s of 5m0s: 444939 executions, 275052 successful
    rsg_test.go:844: 2m35s of 5m0s: 460117 executions, 284469 successful
    rsg_test.go:844: 2m40s of 5m0s: 476688 executions, 294775 successful
    rsg_test.go:844: 2m45s of 5m0s: 492514 executions, 304546 successful
    rsg_test.go:844: 2m50s of 5m0s: 508223 executions, 314290 successful
    rsg_test.go:844: 2m55s of 5m0s: 523829 executions, 323867 successful
    rsg_test.go:844: 3m0s of 5m0s: 537660 executions, 332372 successful
    rsg_test.go:844: 3m5s of 5m0s: 553187 executions, 341986 successful
    rsg_test.go:844: 3m10s of 5m0s: 569107 executions, 351758 successful
    rsg_test.go:844: 3m15s of 5m0s: 583883 executions, 360957 successful
    rsg_test.go:844: 3m20s of 5m0s: 599389 executions, 370531 successful
    rsg_test.go:844: 3m25s of 5m0s: 614639 executions, 379905 successful
    rsg_test.go:844: 3m30s of 5m0s: 628084 executions, 388255 successful
    rsg_test.go:844: 3m35s of 5m0s: 642905 executions, 397519 successful
    rsg_test.go:844: 3m40s of 5m0s: 657683 executions, 406619 successful
    rsg_test.go:844: 3m45s of 5m0s: 673057 executions, 416114 successful
    rsg_test.go:844: 3m50s of 5m0s: 688838 executions, 425846 successful
    rsg_test.go:844: 3m55s of 5m0s: 703864 executions, 435163 successful
    rsg_test.go:844: 4m0s of 5m0s: 717049 executions, 443378 successful
    rsg_test.go:844: 4m5s of 5m0s: 731716 executions, 452351 successful
    rsg_test.go:844: 4m10s of 5m0s: 747189 executions, 461883 successful
    rsg_test.go:844: 4m15s of 5m0s: 762632 executions, 471427 successful
    rsg_test.go:844: 4m20s of 5m0s: 778121 executions, 481070 successful
    rsg_test.go:844: 4m25s of 5m0s: 794334 executions, 491156 successful
    rsg_test.go:844: 4m30s of 5m0s: 807794 executions, 499406 successful
    rsg_test.go:844: 4m35s of 5m0s: 823514 executions, 509114 successful
    rsg_test.go:844: 4m40s of 5m0s: 838427 executions, 518372 successful
    rsg_test.go:844: 4m45s of 5m0s: 853929 executions, 528057 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #107928 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-23.1]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11245463?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11245463?buildTab=artifacts#/) on master @ [a3a389e464defd3d301a0b080435cf33502a02d2](https://github.com/cockroachdb/cockroach/commits/a3a389e464defd3d301a0b080435cf33502a02d2):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 5629566 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:844: 45s of 5m0s: 148836 executions, 92279 successful
    rsg_test.go:844: 50s of 5m0s: 165991 executions, 102877 successful
    rsg_test.go:844: 55s of 5m0s: 183247 executions, 113535 successful
    rsg_test.go:844: 1m0s of 5m0s: 199596 executions, 123755 successful
    rsg_test.go:844: 1m5s of 5m0s: 212484 executions, 131661 successful
    rsg_test.go:844: 1m10s of 5m0s: 228871 executions, 141867 successful
    rsg_test.go:844: 1m15s of 5m0s: 246056 executions, 152482 successful
    rsg_test.go:844: 1m20s of 5m0s: 262710 executions, 162770 successful
    rsg_test.go:844: 1m25s of 5m0s: 279636 executions, 173333 successful
    rsg_test.go:844: 1m30s of 5m0s: 296231 executions, 183614 successful
    rsg_test.go:844: 1m35s of 5m0s: 311594 executions, 193130 successful
    rsg_test.go:844: 1m40s of 5m0s: 328101 executions, 203409 successful
    rsg_test.go:844: 1m45s of 5m0s: 345658 executions, 214305 successful
    rsg_test.go:844: 1m50s of 5m0s: 362702 executions, 224907 successful
    rsg_test.go:844: 1m55s of 5m0s: 380297 executions, 235752 successful
    rsg_test.go:844: 2m0s of 5m0s: 395095 executions, 244937 successful
    rsg_test.go:844: 2m5s of 5m0s: 412120 executions, 255558 successful
    rsg_test.go:844: 2m10s of 5m0s: 429014 executions, 265981 successful
    rsg_test.go:844: 2m15s of 5m0s: 446722 executions, 277014 successful
    rsg_test.go:844: 2m20s of 5m0s: 463082 executions, 287289 successful
    rsg_test.go:844: 2m25s of 5m0s: 480000 executions, 297651 successful
    rsg_test.go:844: 2m30s of 5m0s: 497392 executions, 308423 successful
    rsg_test.go:844: 2m35s of 5m0s: 511932 executions, 317496 successful
    rsg_test.go:844: 2m40s of 5m0s: 528624 executions, 327712 successful
    rsg_test.go:844: 2m45s of 5m0s: 546110 executions, 338586 successful
    rsg_test.go:844: 2m50s of 5m0s: 563414 executions, 349294 successful
    rsg_test.go:844: 2m55s of 5m0s: 580821 executions, 360088 successful
    rsg_test.go:844: 3m0s of 5m0s: 597964 executions, 370808 successful
    rsg_test.go:844: 3m5s of 5m0s: 614462 executions, 380985 successful
    rsg_test.go:844: 3m10s of 5m0s: 629979 executions, 390604 successful
    rsg_test.go:844: 3m15s of 5m0s: 646768 executions, 401028 successful
    rsg_test.go:844: 3m20s of 5m0s: 663461 executions, 411298 successful
    rsg_test.go:844: 3m25s of 5m0s: 681256 executions, 422392 successful
    rsg_test.go:844: 3m30s of 5m0s: 698385 executions, 433006 successful
    rsg_test.go:844: 3m35s of 5m0s: 714426 executions, 442891 successful
    rsg_test.go:844: 3m40s of 5m0s: 729808 executions, 452438 successful
    rsg_test.go:844: 3m45s of 5m0s: 747416 executions, 463329 successful
    rsg_test.go:844: 3m50s of 5m0s: 764671 executions, 474070 successful
    rsg_test.go:844: 3m55s of 5m0s: 782383 executions, 485054 successful
    rsg_test.go:844: 4m0s of 5m0s: 799068 executions, 495415 successful
    rsg_test.go:844: 4m5s of 5m0s: 816709 executions, 506348 successful
    rsg_test.go:844: 4m10s of 5m0s: 831312 executions, 515284 successful
    rsg_test.go:844: 4m15s of 5m0s: 848329 executions, 525843 successful
    rsg_test.go:844: 4m20s of 5m0s: 865265 executions, 536414 successful
    rsg_test.go:844: 4m25s of 5m0s: 882916 executions, 547380 successful
    rsg_test.go:844: 4m30s of 5m0s: 899880 executions, 557803 successful
    rsg_test.go:844: 4m35s of 5m0s: 916002 executions, 567841 successful
    rsg_test.go:844: 4m40s of 5m0s: 932858 executions, 578213 successful
    rsg_test.go:844: 4m45s of 5m0s: 947883 executions, 587506 successful
    rsg_test.go:844: 4m50s of 5m0s: 965028 executions, 598206 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #107928 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-23.1]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11277547?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11277547?buildTab=artifacts#/) on master @ [6fa90e9e1003b0e0355c1ff42eb34b1b64ab36f2](https://github.com/cockroachdb/cockroach/commits/6fa90e9e1003b0e0355c1ff42eb34b1b64ab36f2):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 3850267 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:844: 40s of 5m0s: 119376 executions, 74107 successful
    rsg_test.go:844: 45s of 5m0s: 135487 executions, 84079 successful
    rsg_test.go:844: 50s of 5m0s: 151026 executions, 93683 successful
    rsg_test.go:844: 55s of 5m0s: 167069 executions, 103582 successful
    rsg_test.go:844: 1m0s of 5m0s: 179991 executions, 111683 successful
    rsg_test.go:844: 1m5s of 5m0s: 191026 executions, 118362 successful
    rsg_test.go:844: 1m10s of 5m0s: 204872 executions, 126949 successful
    rsg_test.go:844: 1m15s of 5m0s: 220875 executions, 136922 successful
    rsg_test.go:844: 1m20s of 5m0s: 236816 executions, 146691 successful
    rsg_test.go:844: 1m25s of 5m0s: 253036 executions, 156650 successful
    rsg_test.go:844: 1m30s of 5m0s: 268868 executions, 166409 successful
    rsg_test.go:844: 1m35s of 5m0s: 282462 executions, 174776 successful
    rsg_test.go:844: 1m40s of 5m0s: 298134 executions, 184472 successful
    rsg_test.go:844: 1m45s of 5m0s: 313214 executions, 193743 successful
    rsg_test.go:844: 1m50s of 5m0s: 328875 executions, 203307 successful
    rsg_test.go:844: 1m55s of 5m0s: 343512 executions, 212364 successful
    rsg_test.go:844: 2m0s of 5m0s: 359535 executions, 222349 successful
    rsg_test.go:844: 2m5s of 5m0s: 375592 executions, 232236 successful
    rsg_test.go:844: 2m10s of 5m0s: 388748 executions, 240468 successful
    rsg_test.go:844: 2m15s of 5m0s: 405003 executions, 250435 successful
    rsg_test.go:844: 2m20s of 5m0s: 421222 executions, 260508 successful
    rsg_test.go:844: 2m25s of 5m0s: 436528 executions, 269961 successful
    rsg_test.go:844: 2m30s of 5m0s: 452706 executions, 279965 successful
    rsg_test.go:844: 2m35s of 5m0s: 468937 executions, 290046 successful
    rsg_test.go:844: 2m40s of 5m0s: 482603 executions, 298494 successful
    rsg_test.go:844: 2m45s of 5m0s: 498806 executions, 308466 successful
    rsg_test.go:844: 2m50s of 5m0s: 513250 executions, 317460 successful
    rsg_test.go:844: 2m55s of 5m0s: 529377 executions, 327495 successful
    rsg_test.go:844: 3m0s of 5m0s: 545606 executions, 337478 successful
    rsg_test.go:844: 3m5s of 5m0s: 561371 executions, 347228 successful
    rsg_test.go:844: 3m10s of 5m0s: 574993 executions, 355631 successful
    rsg_test.go:844: 3m15s of 5m0s: 590020 executions, 364985 successful
    rsg_test.go:844: 3m20s of 5m0s: 605816 executions, 374794 successful
    rsg_test.go:844: 3m25s of 5m0s: 621670 executions, 384536 successful
    rsg_test.go:844: 3m30s of 5m0s: 637999 executions, 394616 successful
    rsg_test.go:844: 3m35s of 5m0s: 652703 executions, 403783 successful
    rsg_test.go:844: 3m40s of 5m0s: 669021 executions, 413771 successful
    rsg_test.go:844: 3m45s of 5m0s: 685187 executions, 423652 successful
    rsg_test.go:844: 3m50s of 5m0s: 701191 executions, 433623 successful
    rsg_test.go:844: 3m55s of 5m0s: 717617 executions, 443661 successful
    rsg_test.go:844: 4m0s of 5m0s: 733766 executions, 453693 successful
    rsg_test.go:844: 4m5s of 5m0s: 747642 executions, 462329 successful
    rsg_test.go:844: 4m10s of 5m0s: 763875 executions, 472452 successful
    rsg_test.go:844: 4m15s of 5m0s: 779775 executions, 482116 successful
    rsg_test.go:844: 4m20s of 5m0s: 795810 executions, 492078 successful
    rsg_test.go:844: 4m25s of 5m0s: 812266 executions, 502315 successful
    rsg_test.go:844: 4m30s of 5m0s: 828364 executions, 512303 successful
    rsg_test.go:844: 4m35s of 5m0s: 842432 executions, 520887 successful
    rsg_test.go:844: 4m40s of 5m0s: 858183 executions, 530657 successful
    rsg_test.go:844: 4m45s of 5m0s: 873687 executions, 540241 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #107928 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-23.1]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11290574?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11290574?buildTab=artifacts#/) on master @ [7da4ef5af83c38680c536e7c18a9db61274b9d25](https://github.com/cockroachdb/cockroach/commits/7da4ef5af83c38680c536e7c18a9db61274b9d25):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 5480882 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:844: 45s of 5m0s: 147873 executions, 91575 successful
    rsg_test.go:844: 50s of 5m0s: 165118 executions, 102227 successful
    rsg_test.go:844: 55s of 5m0s: 181319 executions, 112181 successful
    rsg_test.go:844: 1m0s of 5m0s: 195665 executions, 121200 successful
    rsg_test.go:844: 1m5s of 5m0s: 210238 executions, 130210 successful
    rsg_test.go:844: 1m10s of 5m0s: 226227 executions, 140099 successful
    rsg_test.go:844: 1m15s of 5m0s: 243457 executions, 150777 successful
    rsg_test.go:844: 1m20s of 5m0s: 260302 executions, 161279 successful
    rsg_test.go:844: 1m25s of 5m0s: 275709 executions, 170879 successful
    rsg_test.go:844: 1m30s of 5m0s: 292832 executions, 181436 successful
    rsg_test.go:844: 1m35s of 5m0s: 309634 executions, 191766 successful
    rsg_test.go:844: 1m40s of 5m0s: 326587 executions, 202316 successful
    rsg_test.go:844: 1m45s of 5m0s: 342984 executions, 212308 successful
    rsg_test.go:844: 1m50s of 5m0s: 358372 executions, 221821 successful
    rsg_test.go:844: 1m55s of 5m0s: 375146 executions, 232368 successful
    rsg_test.go:844: 2m0s of 5m0s: 391957 executions, 242667 successful
    rsg_test.go:844: 2m5s of 5m0s: 408614 executions, 253078 successful
    rsg_test.go:844: 2m10s of 5m0s: 425240 executions, 263315 successful
    rsg_test.go:844: 2m15s of 5m0s: 442687 executions, 274103 successful
    rsg_test.go:844: 2m20s of 5m0s: 459792 executions, 284646 successful
    rsg_test.go:844: 2m25s of 5m0s: 473492 executions, 293161 successful
    rsg_test.go:844: 2m30s of 5m0s: 490601 executions, 303831 successful
    rsg_test.go:844: 2m35s of 5m0s: 507055 executions, 313962 successful
    rsg_test.go:844: 2m40s of 5m0s: 524170 executions, 324560 successful
    rsg_test.go:844: 2m45s of 5m0s: 541328 executions, 335171 successful
    rsg_test.go:844: 2m50s of 5m0s: 558776 executions, 346046 successful
    rsg_test.go:844: 2m55s of 5m0s: 574175 executions, 355555 successful
    rsg_test.go:844: 3m0s of 5m0s: 590654 executions, 365634 successful
    rsg_test.go:844: 3m5s of 5m0s: 607374 executions, 376045 successful
    rsg_test.go:844: 3m10s of 5m0s: 624756 executions, 386808 successful
    rsg_test.go:844: 3m15s of 5m0s: 641569 executions, 397225 successful
    rsg_test.go:844: 3m20s of 5m0s: 656803 executions, 406665 successful
    rsg_test.go:844: 3m25s of 5m0s: 673895 executions, 417339 successful
    rsg_test.go:844: 3m30s of 5m0s: 690842 executions, 427846 successful
    rsg_test.go:844: 3m35s of 5m0s: 707750 executions, 438252 successful
    rsg_test.go:844: 3m40s of 5m0s: 724720 executions, 448867 successful
    rsg_test.go:844: 3m45s of 5m0s: 739069 executions, 457669 successful
    rsg_test.go:844: 3m50s of 5m0s: 756845 executions, 468719 successful
    rsg_test.go:844: 3m55s of 5m0s: 773571 executions, 479211 successful
    rsg_test.go:844: 4m0s of 5m0s: 790348 executions, 489592 successful
    rsg_test.go:844: 4m5s of 5m0s: 807401 executions, 500164 successful
    rsg_test.go:844: 4m10s of 5m0s: 821835 executions, 509156 successful
    rsg_test.go:844: 4m15s of 5m0s: 839432 executions, 520098 successful
    rsg_test.go:844: 4m20s of 5m0s: 856204 executions, 530485 successful
    rsg_test.go:844: 4m25s of 5m0s: 873240 executions, 541142 successful
    rsg_test.go:844: 4m30s of 5m0s: 889617 executions, 551241 successful
    rsg_test.go:844: 4m35s of 5m0s: 906815 executions, 561919 successful
    rsg_test.go:844: 4m40s of 5m0s: 923077 executions, 572048 successful
    rsg_test.go:844: 4m45s of 5m0s: 937796 executions, 581214 successful
    rsg_test.go:844: 4m50s of 5m0s: 954345 executions, 591407 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #108650 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-23.1.9-rc]
- #107928 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-23.1]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11300096?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11300096?buildTab=artifacts#/) on master @ [6b0f43a13c4d2b36b4da5961e523161dabe212c0](https://github.com/cockroachdb/cockroach/commits/6b0f43a13c4d2b36b4da5961e523161dabe212c0):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 5110721 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:844: 45s of 5m0s: 146083 executions, 90460 successful
    rsg_test.go:844: 50s of 5m0s: 162499 executions, 100662 successful
    rsg_test.go:844: 55s of 5m0s: 179508 executions, 111133 successful
    rsg_test.go:844: 1m0s of 5m0s: 195829 executions, 121299 successful
    rsg_test.go:844: 1m5s of 5m0s: 207992 executions, 128773 successful
    rsg_test.go:844: 1m10s of 5m0s: 223520 executions, 138391 successful
    rsg_test.go:844: 1m15s of 5m0s: 240189 executions, 148773 successful
    rsg_test.go:844: 1m20s of 5m0s: 257062 executions, 159166 successful
    rsg_test.go:844: 1m25s of 5m0s: 273491 executions, 169445 successful
    rsg_test.go:844: 1m30s of 5m0s: 288000 executions, 178341 successful
    rsg_test.go:844: 1m35s of 5m0s: 304393 executions, 188549 successful
    rsg_test.go:844: 1m40s of 5m0s: 321148 executions, 198908 successful
    rsg_test.go:844: 1m45s of 5m0s: 337638 executions, 209118 successful
    rsg_test.go:844: 1m50s of 5m0s: 353858 executions, 219285 successful
    rsg_test.go:844: 1m55s of 5m0s: 370611 executions, 229568 successful
    rsg_test.go:844: 2m0s of 5m0s: 386578 executions, 239481 successful
    rsg_test.go:844: 2m5s of 5m0s: 400385 executions, 248098 successful
    rsg_test.go:844: 2m10s of 5m0s: 417318 executions, 258561 successful
    rsg_test.go:844: 2m15s of 5m0s: 433775 executions, 268825 successful
    rsg_test.go:844: 2m20s of 5m0s: 450169 executions, 278866 successful
    rsg_test.go:844: 2m25s of 5m0s: 466335 executions, 288910 successful
    rsg_test.go:844: 2m30s of 5m0s: 480760 executions, 297886 successful
    rsg_test.go:844: 2m35s of 5m0s: 496523 executions, 307677 successful
    rsg_test.go:844: 2m40s of 5m0s: 513500 executions, 318186 successful
    rsg_test.go:844: 2m45s of 5m0s: 530479 executions, 328730 successful
    rsg_test.go:844: 2m50s of 5m0s: 547155 executions, 339091 successful
    rsg_test.go:844: 2m55s of 5m0s: 561976 executions, 348216 successful
    rsg_test.go:844: 3m0s of 5m0s: 577762 executions, 357999 successful
    rsg_test.go:844: 3m5s of 5m0s: 594280 executions, 368251 successful
    rsg_test.go:844: 3m10s of 5m0s: 610768 executions, 378443 successful
    rsg_test.go:844: 3m15s of 5m0s: 627423 executions, 388859 successful
    rsg_test.go:844: 3m20s of 5m0s: 642344 executions, 398080 successful
    rsg_test.go:844: 3m25s of 5m0s: 658961 executions, 408285 successful
    rsg_test.go:844: 3m30s of 5m0s: 675029 executions, 418221 successful
    rsg_test.go:844: 3m35s of 5m0s: 691636 executions, 428544 successful
    rsg_test.go:844: 3m40s of 5m0s: 708891 executions, 439289 successful
    rsg_test.go:844: 3m45s of 5m0s: 725197 executions, 449325 successful
    rsg_test.go:844: 3m50s of 5m0s: 740948 executions, 459130 successful
    rsg_test.go:844: 3m55s of 5m0s: 757011 executions, 469196 successful
    rsg_test.go:844: 4m0s of 5m0s: 772340 executions, 478659 successful
    rsg_test.go:844: 4m5s of 5m0s: 789346 executions, 489259 successful
    rsg_test.go:844: 4m10s of 5m0s: 805949 executions, 499471 successful
    rsg_test.go:844: 4m15s of 5m0s: 822457 executions, 509785 successful
    rsg_test.go:844: 4m20s of 5m0s: 839079 executions, 520000 successful
    rsg_test.go:844: 4m25s of 5m0s: 856204 executions, 530601 successful
    rsg_test.go:844: 4m30s of 5m0s: 870518 executions, 539490 successful
    rsg_test.go:844: 4m35s of 5m0s: 887943 executions, 550259 successful
    rsg_test.go:844: 4m40s of 5m0s: 904926 executions, 560765 successful
    rsg_test.go:844: 4m45s of 5m0s: 921804 executions, 571309 successful
    rsg_test.go:844: 4m50s of 5m0s: 938991 executions, 581956 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #108650 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-23.1.9-rc]
- #107928 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-23.1]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11308693?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11308693?buildTab=artifacts#/) on master @ [c13bf7633cbb416d9e43f8c57b1e309fab1110ce](https://github.com/cockroachdb/cockroach/commits/c13bf7633cbb416d9e43f8c57b1e309fab1110ce):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 3783306 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:844: 40s of 5m0s: 116747 executions, 72131 successful
    rsg_test.go:844: 45s of 5m0s: 133158 executions, 82384 successful
    rsg_test.go:844: 50s of 5m0s: 148786 executions, 92033 successful
    rsg_test.go:844: 55s of 5m0s: 163924 executions, 101332 successful
    rsg_test.go:844: 1m0s of 5m0s: 178194 executions, 110199 successful
    rsg_test.go:844: 1m5s of 5m0s: 189842 executions, 117437 successful
    rsg_test.go:844: 1m10s of 5m0s: 202564 executions, 125280 successful
    rsg_test.go:844: 1m15s of 5m0s: 217821 executions, 134647 successful
    rsg_test.go:844: 1m20s of 5m0s: 233446 executions, 144333 successful
    rsg_test.go:844: 1m25s of 5m0s: 249113 executions, 153974 successful
    rsg_test.go:844: 1m30s of 5m0s: 264667 executions, 163550 successful
    rsg_test.go:844: 1m35s of 5m0s: 278895 executions, 172365 successful
    rsg_test.go:844: 1m40s of 5m0s: 294352 executions, 181981 successful
    rsg_test.go:844: 1m45s of 5m0s: 310111 executions, 191654 successful
    rsg_test.go:844: 1m50s of 5m0s: 326283 executions, 201794 successful
    rsg_test.go:844: 1m55s of 5m0s: 341981 executions, 211514 successful
    rsg_test.go:844: 2m0s of 5m0s: 356795 executions, 220681 successful
    rsg_test.go:844: 2m5s of 5m0s: 370643 executions, 229221 successful
    rsg_test.go:844: 2m10s of 5m0s: 386205 executions, 238909 successful
    rsg_test.go:844: 2m15s of 5m0s: 401985 executions, 248692 successful
    rsg_test.go:844: 2m20s of 5m0s: 418051 executions, 258669 successful
    rsg_test.go:844: 2m25s of 5m0s: 433841 executions, 268365 successful
    rsg_test.go:844: 2m30s of 5m0s: 447987 executions, 276984 successful
    rsg_test.go:844: 2m35s of 5m0s: 463758 executions, 286686 successful
    rsg_test.go:844: 2m40s of 5m0s: 479742 executions, 296684 successful
    rsg_test.go:844: 2m45s of 5m0s: 495710 executions, 306596 successful
    rsg_test.go:844: 2m50s of 5m0s: 511423 executions, 316266 successful
    rsg_test.go:844: 2m55s of 5m0s: 527434 executions, 326234 successful
    rsg_test.go:844: 3m0s of 5m0s: 541494 executions, 334893 successful
    rsg_test.go:844: 3m5s of 5m0s: 557561 executions, 344736 successful
    rsg_test.go:844: 3m10s of 5m0s: 572807 executions, 354164 successful
    rsg_test.go:844: 3m15s of 5m0s: 589160 executions, 364287 successful
    rsg_test.go:844: 3m20s of 5m0s: 605058 executions, 374071 successful
    rsg_test.go:844: 3m25s of 5m0s: 621238 executions, 384163 successful
    rsg_test.go:844: 3m30s of 5m0s: 635125 executions, 392698 successful
    rsg_test.go:844: 3m35s of 5m0s: 651685 executions, 402879 successful
    rsg_test.go:844: 3m40s of 5m0s: 666764 executions, 412185 successful
    rsg_test.go:844: 3m45s of 5m0s: 681835 executions, 421546 successful
    rsg_test.go:844: 3m50s of 5m0s: 697151 executions, 430988 successful
    rsg_test.go:844: 3m55s of 5m0s: 712949 executions, 440744 successful
    rsg_test.go:844: 4m0s of 5m0s: 728870 executions, 450630 successful
    rsg_test.go:844: 4m5s of 5m0s: 742896 executions, 459197 successful
    rsg_test.go:844: 4m10s of 5m0s: 759268 executions, 469365 successful
    rsg_test.go:844: 4m15s of 5m0s: 774547 executions, 478755 successful
    rsg_test.go:844: 4m20s of 5m0s: 790220 executions, 488381 successful
    rsg_test.go:844: 4m25s of 5m0s: 805720 executions, 497930 successful
    rsg_test.go:844: 4m30s of 5m0s: 821717 executions, 507762 successful
    rsg_test.go:844: 4m35s of 5m0s: 835690 executions, 516420 successful
    rsg_test.go:844: 4m40s of 5m0s: 850944 executions, 525783 successful
    rsg_test.go:844: 4m45s of 5m0s: 867186 executions, 535916 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #108650 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-23.1.9-rc]
- #107928 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-23.1]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11324159?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11324159?buildTab=artifacts#/) on master @ [abf61bbf38a990f938b56a80284a834e2d087ea1](https://github.com/cockroachdb/cockroach/commits/abf61bbf38a990f938b56a80284a834e2d087ea1):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 4888116 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:844: 45s of 5m0s: 144519 executions, 89584 successful
    rsg_test.go:844: 50s of 5m0s: 160680 executions, 99587 successful
    rsg_test.go:844: 55s of 5m0s: 177541 executions, 110025 successful
    rsg_test.go:844: 1m0s of 5m0s: 193826 executions, 120113 successful
    rsg_test.go:844: 1m5s of 5m0s: 207927 executions, 128787 successful
    rsg_test.go:844: 1m10s of 5m0s: 221732 executions, 137335 successful
    rsg_test.go:844: 1m15s of 5m0s: 238006 executions, 147454 successful
    rsg_test.go:844: 1m20s of 5m0s: 254418 executions, 157661 successful
    rsg_test.go:844: 1m25s of 5m0s: 270959 executions, 167870 successful
    rsg_test.go:844: 1m30s of 5m0s: 287588 executions, 178187 successful
    rsg_test.go:844: 1m35s of 5m0s: 304502 executions, 188648 successful
    rsg_test.go:844: 1m40s of 5m0s: 321271 executions, 198960 successful
    rsg_test.go:844: 1m45s of 5m0s: 335445 executions, 207754 successful
    rsg_test.go:844: 1m50s of 5m0s: 351989 executions, 218021 successful
    rsg_test.go:844: 1m55s of 5m0s: 368710 executions, 228317 successful
    rsg_test.go:844: 2m0s of 5m0s: 385208 executions, 238587 successful
    rsg_test.go:844: 2m5s of 5m0s: 402567 executions, 249370 successful
    rsg_test.go:844: 2m10s of 5m0s: 419199 executions, 259560 successful
    rsg_test.go:844: 2m15s of 5m0s: 432828 executions, 267983 successful
    rsg_test.go:844: 2m20s of 5m0s: 449606 executions, 278427 successful
    rsg_test.go:844: 2m25s of 5m0s: 466566 executions, 288932 successful
    rsg_test.go:844: 2m30s of 5m0s: 482785 executions, 299015 successful
    rsg_test.go:844: 2m35s of 5m0s: 500055 executions, 309626 successful
    rsg_test.go:844: 2m40s of 5m0s: 516470 executions, 319737 successful
    rsg_test.go:844: 2m45s of 5m0s: 532030 executions, 329401 successful
    rsg_test.go:844: 2m50s of 5m0s: 548042 executions, 339328 successful
    rsg_test.go:844: 2m55s of 5m0s: 564210 executions, 349315 successful
    rsg_test.go:844: 3m0s of 5m0s: 581389 executions, 359939 successful
    rsg_test.go:844: 3m5s of 5m0s: 598258 executions, 370380 successful
    rsg_test.go:844: 3m10s of 5m0s: 614640 executions, 380543 successful
    rsg_test.go:844: 3m15s of 5m0s: 631395 executions, 390849 successful
    rsg_test.go:844: 3m20s of 5m0s: 645890 executions, 399816 successful
    rsg_test.go:844: 3m25s of 5m0s: 663180 executions, 410570 successful
    rsg_test.go:844: 3m30s of 5m0s: 679962 executions, 420987 successful
    rsg_test.go:844: 3m35s of 5m0s: 696578 executions, 431261 successful
    rsg_test.go:844: 3m40s of 5m0s: 713559 executions, 441725 successful
    rsg_test.go:844: 3m45s of 5m0s: 730727 executions, 452382 successful
    rsg_test.go:844: 3m50s of 5m0s: 747192 executions, 462518 successful
    rsg_test.go:844: 3m55s of 5m0s: 761616 executions, 471567 successful
    rsg_test.go:844: 4m0s of 5m0s: 777541 executions, 481470 successful
    rsg_test.go:844: 4m5s of 5m0s: 794415 executions, 491883 successful
    rsg_test.go:844: 4m10s of 5m0s: 811764 executions, 502614 successful
    rsg_test.go:844: 4m15s of 5m0s: 828884 executions, 513193 successful
    rsg_test.go:844: 4m20s of 5m0s: 844897 executions, 523156 successful
    rsg_test.go:844: 4m25s of 5m0s: 859381 executions, 532205 successful
    rsg_test.go:844: 4m30s of 5m0s: 875414 executions, 542087 successful
    rsg_test.go:844: 4m35s of 5m0s: 892051 executions, 552335 successful
    rsg_test.go:844: 4m40s of 5m0s: 909233 executions, 562947 successful
    rsg_test.go:844: 4m45s of 5m0s: 926187 executions, 573401 successful
    rsg_test.go:844: 4m50s of 5m0s: 942551 executions, 583540 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #108650 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-23.1.9-rc]
- #107928 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-23.1]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestRandomSyntaxFunctions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11339549?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/11339549?buildTab=artifacts#/) on master @ [e0de1f12d4f496fc7e4050a33e96f8c635c7a27c](https://github.com/cockroachdb/cockroach/commits/e0de1f12d4f496fc7e4050a33e96f8c635c7a27c):

Fatal error:

```
panic: test timed out after 14m55s

```
Stack: 

```
goroutine 3879452 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2036 +0x8e
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x32
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    rsg_test.go:844: 40s of 5m0s: 122027 executions, 75316 successful
    rsg_test.go:844: 45s of 5m0s: 137295 executions, 84862 successful
    rsg_test.go:844: 50s of 5m0s: 152629 executions, 94361 successful
    rsg_test.go:844: 55s of 5m0s: 168671 executions, 104289 successful
    rsg_test.go:844: 1m0s of 5m0s: 181926 executions, 112513 successful
    rsg_test.go:844: 1m5s of 5m0s: 193356 executions, 119510 successful
    rsg_test.go:844: 1m10s of 5m0s: 206517 executions, 127631 successful
    rsg_test.go:844: 1m15s of 5m0s: 222391 executions, 137451 successful
    rsg_test.go:844: 1m20s of 5m0s: 237681 executions, 146937 successful
    rsg_test.go:844: 1m25s of 5m0s: 253352 executions, 156638 successful
    rsg_test.go:844: 1m30s of 5m0s: 267029 executions, 165073 successful
    rsg_test.go:844: 1m35s of 5m0s: 282802 executions, 174815 successful
    rsg_test.go:844: 1m40s of 5m0s: 297883 executions, 184122 successful
    rsg_test.go:844: 1m45s of 5m0s: 313790 executions, 194042 successful
    rsg_test.go:844: 1m50s of 5m0s: 329171 executions, 203544 successful
    rsg_test.go:844: 1m55s of 5m0s: 343999 executions, 212771 successful
    rsg_test.go:844: 2m0s of 5m0s: 359110 executions, 222005 successful
    rsg_test.go:844: 2m5s of 5m0s: 375045 executions, 231896 successful
    rsg_test.go:844: 2m10s of 5m0s: 391481 executions, 241962 successful
    rsg_test.go:844: 2m15s of 5m0s: 406562 executions, 251419 successful
    rsg_test.go:844: 2m20s of 5m0s: 422599 executions, 261218 successful
    rsg_test.go:844: 2m25s of 5m0s: 436436 executions, 269792 successful
    rsg_test.go:844: 2m30s of 5m0s: 452725 executions, 279847 successful
    rsg_test.go:844: 2m35s of 5m0s: 468978 executions, 289975 successful
    rsg_test.go:844: 2m40s of 5m0s: 484562 executions, 299590 successful
    rsg_test.go:844: 2m45s of 5m0s: 500447 executions, 309398 successful
    rsg_test.go:844: 2m50s of 5m0s: 516719 executions, 319392 successful
    rsg_test.go:844: 2m55s of 5m0s: 530158 executions, 327689 successful
    rsg_test.go:844: 3m0s of 5m0s: 546288 executions, 337662 successful
    rsg_test.go:844: 3m5s of 5m0s: 561851 executions, 347402 successful
    rsg_test.go:844: 3m10s of 5m0s: 578222 executions, 357634 successful
    rsg_test.go:844: 3m15s of 5m0s: 592658 executions, 366541 successful
    rsg_test.go:844: 3m20s of 5m0s: 608013 executions, 375987 successful
    rsg_test.go:844: 3m25s of 5m0s: 624118 executions, 386053 successful
    rsg_test.go:844: 3m30s of 5m0s: 637027 executions, 394075 successful
    rsg_test.go:844: 3m35s of 5m0s: 653450 executions, 404154 successful
    rsg_test.go:844: 3m40s of 5m0s: 669513 executions, 414071 successful
    rsg_test.go:844: 3m45s of 5m0s: 685065 executions, 423652 successful
    rsg_test.go:844: 3m50s of 5m0s: 700135 executions, 432964 successful
    rsg_test.go:844: 3m55s of 5m0s: 715906 executions, 442712 successful
    rsg_test.go:844: 4m0s of 5m0s: 729711 executions, 451261 successful
    rsg_test.go:844: 4m5s of 5m0s: 745371 executions, 461014 successful
    rsg_test.go:844: 4m10s of 5m0s: 761095 executions, 470778 successful
    rsg_test.go:844: 4m15s of 5m0s: 776613 executions, 480400 successful
    rsg_test.go:844: 4m20s of 5m0s: 792265 executions, 490046 successful
    rsg_test.go:844: 4m25s of 5m0s: 806608 executions, 499034 successful
    rsg_test.go:844: 4m30s of 5m0s: 820804 executions, 507708 successful
    rsg_test.go:844: 4m35s of 5m0s: 836571 executions, 517401 successful
    rsg_test.go:844: 4m40s of 5m0s: 852439 executions, 527163 successful
    rsg_test.go:844: 4m45s of 5m0s: 868298 executions, 536958 successful
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #108650 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-23.1.9-rc]
- #107928 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-23.1]
- #95619 sql/tests: TestRandomSyntaxFunctions failed [C-test-failure O-robot T-sql-foundations branch-release-22.2]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxFunctions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/create_stats.go
+++ b/pkg/sql/create_stats.go
@@ -84,7 +84,7 @@ func StubTableStats(
 }
 
 // createStatsNode is a planNode implemented in terms of a function. The
-// startJob function starts a Job during Start, and the remainder of the
+// runJob function starts a Job during Start, and the remainder of the
 // CREATE STATISTICS planning and execution is performed within the jobs
 // framework.
 type createStatsNode struct {
@@ -97,44 +97,23 @@ type createStatsNode struct {
 	// If it is false, the flow for create statistics is planned directly; this
 	// is used when the statement is under EXPLAIN or EXPLAIN ANALYZE.
 	runAsJob bool
-
-	run createStatsRun
-}
-
-// createStatsRun contains the run-time state of createStatsNode during local
-// execution.
-type createStatsRun struct {
-	errCh chan error
 }
 
 func (n *createStatsNode) startExec(params runParams) error {
 	telemetry.Inc(sqltelemetry.SchemaChangeCreateCounter("stats"))
-	n.run.errCh = make(chan error)
-	go func() {
-		err := n.startJob(params.ctx)
-		select {
-		case <-params.ctx.Done():
-		case n.run.errCh <- err:
-		}
-		close(n.run.errCh)
-	}()
-	return nil
+	return n.runJob(params.ctx)
 }
 
 func (n *createStatsNode) Next(params runParams) (bool, error) {
-	select {
-	case <-params.ctx.Done():
-		return false, params.ctx.Err()
-	case err := <-n.run.errCh:
-		return false, err
-	}
+	return false, nil
 }
 
 func (*createStatsNode) Close(context.Context) {}
 func (*createStatsNode) Values() tree.Datums   { return nil }
 
-// startJob starts a CreateStats job to plan and execute statistics creation.
-func (n *createStatsNode) startJob(ctx context.Context) error {
+// runJob starts a CreateStats job synchronously to plan and execute
+// statistics creation and then waits for the job to complete.
+func (n *createStatsNode) runJob(ctx context.Context) error {
 	record, err := n.makeJobRecord(ctx)
 	if err != nil {
 		return err
@@ -167,8 +146,7 @@ func (n *createStatsNode) startJob(ctx context.Context) error {
 	if err := job.Start(ctx); err != nil {
 		return err
 	}
-
-	if err := job.AwaitCompletion(ctx); err != nil {
+	if err = job.AwaitCompletion(ctx); err != nil {
 		if errors.Is(err, stats.ConcurrentCreateStatsError) {
 			// Delete the job so users don't see it and get confused by the error.
 			const stmt = `DELETE FROM system.jobs WHERE id = $1`
@@ -178,9 +156,8 @@ func (n *createStatsNode) startJob(ctx context.Context) error {
 				log.Warningf(ctx, "failed to delete job: %v", delErr)
 			}
 		}
-		return err
 	}
-	return nil
+	return err
 }
 
 // makeJobRecord creates a CreateStats job record which can be used to plan and

--- a/pkg/sql/stats/create_stats_job_test.go
+++ b/pkg/sql/stats/create_stats_job_test.go
@@ -29,6 +29,8 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/rowexec"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/jobutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util/encoding"
@@ -36,6 +38,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/retry"
 	"github.com/cockroachdb/errors"
+	"github.com/stretchr/testify/require"
 )
 
 // TestCreateStatsControlJob tests that PAUSE JOB, RESUME JOB, and CANCEL JOB
@@ -120,6 +123,55 @@ func TestCreateStatsControlJob(t *testing.T) {
 				{"s2", "{x}", "1000"},
 			})
 	})
+}
+
+func TestCreateStatisticsCanBeCancelled(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+	skip.UnderStress(t, "test can be slow to quiesce because of filter")
+
+	var allowRequest chan struct{}
+
+	var serverArgs base.TestServerArgs
+	filter, setTableID := createStatsRequestFilter(&allowRequest)
+	serverArgs.Knobs.JobsTestingKnobs = jobs.NewTestingKnobsWithShortIntervals()
+	serverArgs.Knobs.Store = &kvserver.StoreTestingKnobs{
+		TestingRequestFilter: filter,
+	}
+
+	ctx := context.Background()
+	tc, conn, _ := serverutils.StartServer(t, serverArgs)
+	defer tc.Stopper().Stop(ctx)
+	sqlDB := sqlutils.MakeSQLRunner(conn)
+
+	sqlDB.Exec(t, `CREATE DATABASE d`)
+	sqlDB.Exec(t, `CREATE TABLE d.t (x INT PRIMARY KEY)`)
+	sqlDB.Exec(t, `INSERT INTO d.t SELECT generate_series(1,1000)`)
+	var tID descpb.ID
+	sqlDB.QueryRow(t, `SELECT 'd.t'::regclass::int`).Scan(&tID)
+	setTableID(tID)
+
+	// Run CREATE STATISTICS and wait for to create the job.
+	allowRequest = make(chan struct{})
+	errCh := make(chan error)
+	go func() {
+		_, err := conn.Exec(`CREATE STATISTICS s1 FROM d.t`)
+		errCh <- err
+	}()
+	allowRequest <- struct{}{}
+	testutils.SucceedsSoon(t, func() error {
+		row := conn.QueryRow("SELECT query_id FROM [SHOW CLUSTER STATEMENTS] WHERE query LIKE 'CREATE STATISTICS%';")
+		var queryID string
+		if err := row.Scan(&queryID); err != nil {
+			return err
+		}
+		_, err := conn.Exec("CANCEL QUERIES VALUES ((SELECT query_id FROM [SHOW CLUSTER STATEMENTS] WHERE query LIKE 'CREATE STATISTICS%'));")
+		return err
+	})
+	err := <-errCh
+	allowRequest <- struct{}{}
+
+	require.ErrorContains(t, err, "pq: query execution canceled")
 }
 
 func TestAtMostOneRunningCreateStats(t *testing.T) {

--- a/pkg/sql/tests/rsg_test.go
+++ b/pkg/sql/tests/rsg_test.go
@@ -173,6 +173,7 @@ func (db *verifyFormatDB) execWithResettableTimeout(
 	}()
 	retry := true
 	targetDuration := duration
+	cancellationChannel := ctx.Done()
 	for retry {
 		retry = false
 		err := func() error {
@@ -200,7 +201,7 @@ func (db *verifyFormatDB) execWithResettableTimeout(
 					return &nonCrasher{sql: sql, err: err}
 				}
 				return nil
-			case <-ctx.Done():
+			case <-cancellationChannel:
 				// Sanity: The context is cancelled when the test is about to
 				// timeout. We will log whatever statement we're waiting on for
 				// debugging purposes. Sometimes queries won't respect
@@ -209,6 +210,7 @@ func (db *verifyFormatDB) execWithResettableTimeout(
 				// We will intentionally retry, which will us to wait for the
 				// go routine to complete above to avoid leaking it.
 				retry = true
+				cancellationChannel = nil
 				return nil
 			case <-time.After(targetDuration):
 				db.mu.Lock()
@@ -356,6 +358,10 @@ func TestRandomSyntaxFunctions(t *testing.T) {
 				switch lower {
 				case "pg_sleep":
 					continue
+				case "crdb_internal.create_sql_schema_telemetry_job":
+					// We can create a crazy number of telemtry jobs accidentally,
+					// within the test. Leading to terrible contention.
+					continue
 				case "crdb_internal.gen_rand_ident":
 					// Generates random identifiers, so a large number are dangerous and
 					// can take a long time.
@@ -435,6 +441,7 @@ func TestRandomSyntaxFunctions(t *testing.T) {
 		// involve schema changes like truncates. In general this should make
 		// this test more resilient as the timeouts are reset as long progress
 		// is made on *some* connection.
+		t.Logf("Running %q", s)
 		return db.execWithResettableTimeout(t, ctx, s, *flagRSGExecTimeout, *flagRSGGoRoutines)
 	})
 }

